test/functional/import-abort-rescan.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test wallet import RPCs.
+
+Test rescan behavior of importprivkey when aborted. The test ensures that:
+1. The abortrescan command indeed stops the rescan process.
+2. Subsequent rescan catches the aborted address UTXO
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (assert_equal, get_rpc_proxy)
+from decimal import Decimal
+import threading # for bg importprivkey
+import time      # for sleep
+
+class ImportAbortRescanTest(BitcoinTestFramework):
+    def __init__(self):
+        super().__init__()
+        # self.setup_clean_chain = False
+
+    def run_test(self):
+        # Generate for BTC
+        self.nodes[0].generate(300)
+        self.log.info("Creating blocks with transactions ...")
+        # Make blocks with spam to cause rescan delay
+        for i in range(200):
+            if i % 50 == 0: self.log.info("... %2.f%%", 100.0 / 200.0 * i)
```

Comment:
nit: if block should appear on the next line. Same for the `if xxxxx: break` lines below